src/import_edf.jl

```diff
@@ -172,10 +176,12 @@ converted to Onda units and checked for consistency, and a promoted encoding
 
 No conversion of the actual signals is performed at this step.
 """
-function edf_signals_to_samplesinfo(edf::EDF.File, edf_signals::Vector{<:EDF.Signal}, kind, channel_names)
-    onda_units = map(s -> edf_to_onda_unit(s.header.physical_dimension), edf_signals)
-    onda_sample_unit = first(onda_units)
-    all(==(onda_sample_unit), onda_units) || error("multiple possible units found for same signal: $onda_units")
+function edf_signals_to_samplesinfo(edf::EDF.File, edf_signals::Vector{<:EDF.Signal}, kind, channel_names; unit_alternatives=STANDARD_UNITS)
+    onda_units = map(s -> edf_to_onda_unit(s.header.physical_dimension, unit_alternatives), edf_signals)
+    onda_sample_units = Set(u for u in onda_units if u != "unknown")
+    onda_sample_units = isempty(onda_sample_units) ? Set(["unknown"]) : onda_sample_units
+    length(onda_sample_units) == 1 || error("multiple possible units found for same signal: $onda_sample_units")
+    onda_sample_unit = first(onda_sample_units)
```

Comment:
Add something to the docstring about this behavior?

Reply:
I'm not sure this behavior is desirable in practice...it seems like a major potential footgun to silently and automatically assume that, say, an EMG signal is in microvolts just because the other EMG signals are since EMG signals are also often in volts...  I don't know enough about these setups to know whether it's even possible that you'd get multi-unit setups for the same signal type AND have unknown units on top of that, so may be worth checking with some domain experts.

The alternative would be to error (the existing behavior) or to discard the "unknowns".

Reply:
An alternative design could be to define Errors containing fields that capture what the error is about (so for example, the vector of parsed channels and corresponding units), and letting users pass in an `on_error` callable that handles these, either by re-throwing or by returning something. I kinda like that.